go-controller/pkg/node/gateway_localnet_linux_test.go

```diff
@@ -281,7 +281,7 @@ var _ = Describe("Node Operations", func() {
 				)
```

Comment:
hi @tssurya . I seem to be consistently failing at this test, when doing a `make test 2>&1 | tee ~/x` from the go-controller directory. Do you see the same? Interestingly, I do not see it failing if I run that test alone, so it might be a cleanup/timing issue when it runs with the other tests.

```
...
I0110 20:32:52.677008   12831 gateway_shared_intf.go:546] Adding service service1 in namespace namespace1^M
I0110 20:32:52.677288   12831 gateway_shared_intf.go:567] Rules already programmed for service1 in namespace namespace1^M
I0110 20:32:52.677396   12831 factory.go:568] Added *v1.Service event handler 1^M
I0110 20:32:52.677479   12831 gateway_shared_intf.go:546] Adding service service1 in namespace namespace1^M
I0110 20:32:52.677681   12831 gateway_shared_intf.go:567] Rules already programmed for service1 in namespace namespace1^M
^M
^[[91m^[[1m• Failure [0.192 seconds]^[[0m^M
Node Operations^M
^[[90m/go/src/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/gateway_localnet_linux_test.go:219^[[0m^M
  on add^M
  ^[[90m/go/src/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/gateway_localnet_linux_test.go:377^[[0m^M
    ^[[91m^[[1minits iptables rules and openflows with LoadBalancer where AllocateLoadBalancerNodePorts=False, ETP=local, LGW mode [It]^[[0m^M
    ^[[90m/go/src/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/gateway_localnet_linux_test.go:837^[[0m^M
^M
    ^[[91mExpected success, but got an error:^M
        <*errors.errorString | 0xc000111af0>: {^M
            s: "expected 0 [] rules in chain nat/OVN-KUBE-EGRESS-SVC, got 1 [-m mark --mark 0x3f0 -m comment --comment Do not SNAT to SVC VIP -j RETURN]",^M
        }^M
        expected 0 [] rules in chain nat/OVN-KUBE-EGRESS-SVC, got 1 [-m mark --mark 0x3f0 -m comment --comment Do not SNAT to SVC VIP -j RETURN]^[[0m^M
^M
    /go/src/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/node/gateway_localnet_linux_test.go:957^M
 ...

[Fail] Node Operations on add [It] inits iptables rules and openflows with LoadBalancer where AllocateLoadBalancerNodePorts=False, ETP=local, LGW mode
```

Reply:
oh right! sorry actually I probably need to rebase on top of master, you see my other bugfix which got in 2 days ago: https://github.com/ovn-org/ovn-kubernetes/pull/3151 adds a new test which needs this tweak as well and that is the reason why this hasn't failed CI here.

I have updated this locally. Will push it with other review comments. Nice Catch!

Reply:
fixed.